src/Solr/Query/SearchQueryBuilder.php

```diff
@@ -236,4 +239,28 @@ protected function getEventDispatcher()
     {
         return $this->eventDispatcher;
     }
+
+    /**
+     * Don't use "E" notation for values lower than 0.00001 (i.e. use 0.0000001 instead of 1E-07)
+     * Taken from https://stackoverflow.com/a/10917464/3141504
+     *
+     * @param float $boost
+     * @return string
+     */
+    private function getNormalizedBoost($boost)
+    {
+        if ($boost == 0) {
+            return $boost;
+        }
+        if ($boost < 1) {
+            $boost = rtrim(
+                rtrim(
+                    sprintf('%.20F', $boost),
+                    '0'
+                ),
+                '.'
```

Comment:
a single `rtrim(..., '0.')` should do the same

Reply:
(if trimming is necessary)

Reply:
No, it doesn't, it would trim 20 to 2. That's why we have two `rtrim`s.